src/hotspot/share/oops/klassVtable.cpp

```diff
@@ -225,15 +222,15 @@ void klassVtable::initialize_vtable(bool checkconstraints, TRAPS) {
           assert(def_vtable_indices->length() == len, "reinit vtable len?");
         }
         for (int i = 0; i < len; i++) {
-          HandleMark hm(THREAD);
-          assert(default_methods->at(i)->is_method(), "must be a Method*");
           methodHandle mh(THREAD, default_methods->at(i));
```

Comment:
Maybe wrap the method in a {} block so you can see when it goes out of scope and don't get tempted to use it after it becomes invalid.

Reply:
Do you mean this methodHandle mh?  vs the refetch of method below on line 232?  Since your main comment was to not mix the behavior change with the cleanup, I'm going to revert line 232 and do a different PR for that so that I can adequately describe the redefinition problem.